modules/local/bwamem2_mem.nf

```diff
@@ -1,29 +1,38 @@
-params.bwa_options = "-K 100000000 -M"
-params.sequencer = "ILLUMINA"
-
 process BWAMEM2_MEM {
-    label 'CPUS_MAX'
-
-    tag "${sample}_${run}"
+    tag "${meta.id}"
+    label 'process_high'
 
-    publishDir "${params.outdir}/bwamem2_mem", mode: 'copy'
+    publishDir "${params.outdir}/bwamem2/${meta.sample}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename ->
+                    if (filename.endsWith('.version.txt')) null
+                    else filename }
 
     input:
-        tuple val(patient), val(sample), val(run), path(read1), path(read2)
+        tuple val(meta), path(reads)
         path bwa
         path fasta
         path fai
+        val options
 
     output:
-        tuple val(patient), val(sample), val(run), path("*.bam"), path("*.bai")
+        tuple val(meta), path("*.bam"), path("*.bai")
 
     script:
     CN = params.sequencing_center ? "CN:${params.sequencing_center}\\t" : ""
-    readGroup = "@RG\\tID:${run}\\t${CN}PU:${run}\\tSM:${sample}\\tLB:${sample}\\tPL:${params.sequencer}"
+    readGroup = "@RG\\tID:${meta.run}\\t${CN}PU:${meta.run}\\tSM:${meta.sample}\\tLB:${meta.sample}\\tPL:ILLUMINA"
```

Comment:
This should come in via the `meta` parameter because not everyone will want to create the read group in this way because they wont have all of the same values in map e.g. see [here](https://github.com/drpatelh/chipseq/blob/7bd0e6b0e619754845e2f65bb4641671bba42bdf/modules/local/process/check_samplesheet.nf#L28-L36)